components/scream/src/physics/cld_fraction/atmosphere_cldfraction.cpp

```diff
@@ -0,0 +1,157 @@
+#include "atmosphere_cldfraction.hpp"
+
+#include "ekat/ekat_assert.hpp"
+#include "ekat/util/ekat_units.hpp"
+
+#include <array>
+
+namespace scream
+{
+  using namespace cldfrac;
+// =========================================================================================
+CldFraction::CldFraction (const ekat::Comm& comm, const ekat::ParameterList& params)
+ : m_cldfraction_comm (comm)
+ , m_cldfraction_params (params)
+{
+  // Nothing to do here
+}
+
+// =========================================================================================
+void CldFraction::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace ekat::units;
+
+  // The units of mixing ratio Q are technically non-dimensional.
+  // Nevertheless, for output reasons, we like to see 'kg/kg'.
+  auto Q = kg/kg;
+  Q.set_string("kg/kg");
+  Units nondim(0,0,0,0,0,0,0);
+
+  const auto& grid_name = m_cldfraction_params.get<std::string>("Grid");
+  auto grid  = grids_manager->get_grid(grid_name);
+  m_num_cols = grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = grid->get_num_vertical_levels();  // Number of levels per column
+  m_nk_pack  = ekat::npack<Spack>(m_num_levs);
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and interfaces 
+  FieldLayout scalar3d_layout_mid { {COL,LEV}, {m_num_cols,m_num_levs} };
+
+  // Set of fields used strictly as input
+  m_required_fields.emplace("qi",   scalar3d_layout_mid, Q,      grid_name);
+  m_required_fields.emplace("alst", scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used strictly as output
+  m_computed_fields.emplace("ast",   scalar3d_layout_mid, nondim, grid_name);
+  m_computed_fields.emplace("aist",  scalar3d_layout_mid, nondim, grid_name);
```

Comment:
Can we use cld_frac_l, cld_frac_i, and cld_frac_tot as liquid, ice, and total cloud fractions rather than aist, alst, and ast?

Reply:
I vote for `liq_cld_frac`, `ice_cld_frac`, and `tot_cld_frac`.

Reply:
I would vote for a name that includes the words `liq` and `ice` instead of `l` and `i` since they are more descriptive. 

But more to the point of this PR, changing the field managed name would require us to change the FM name in SHOC and P3, and update the initial condition files for their stand-alone tests.  All of which I am happy to do, but do we want that to be in this PR, or a subsequent PR?

Alternatively, we establish these names here just for cld_fraction and then update shoc and p3 interfaces when it comes time to couple them.

@bartgol , @PeterCaldwell thoughts?

Reply:
Well, I don't think this PR should change fields names in other parts of the code. That said, I don't think there's any test that uses both shoc/p3 _and_ IceCldFraction, right? So whatever name you pick here, is not going to cause conflicts.

But I guess it's ok to keep the same name as shoc/p3, and change it in the name-changing PR.

Reply:
Yeah, I like Luca's names much better than mine. Maybe cld_frac_liq and cld_frac_ice, though, to emphasize that these are all cloud fractions first and flavors next. This is consistent with the naming convention for qi, qc, nc, ni, etc.

Reply:
Yeah, maybe the easiest thing to do is to just leave (N)ast(Y) cld frac names in this PR and to do a subsequent PR that just switches everything to cld_frac_ice and so on later. Though like Luca says, naming conventions are compartmentalized for each proc, so they don't need to be consistent.

Reply:
I'm in favor of changing the variable names to the better names and leave the field managed names for the cleanup we already see on the horizon.  That way coupling between this process and others is immediately possible, and the task of cleaning up the field managed name won't delay progress anywhere else.